tests/test_metrics.py

```diff
@@ -51,17 +51,32 @@ def _testMatchedRefCoaddMetricDerived(self, type_metric: type[MatchedRefCoaddMet
             self.assertGreater(len(tester(data)), 0)
 
     def testMatchedRefCoaddMetric(self):
-        tester = MatchedRefCoaddMetric(unit="")
+        kwargs = {key: "" for key in ("unit", "name_prefix")}
+        for kwarg in kwargs:
+            kwargs_init = {kwarg: ""}
+            tester = MatchedRefCoaddMetric(**kwargs_init)
+            # Validation will fail because configureMetrics hasn't been called
+            with self.assertRaises(ValueError):
+                tester.validate()
+            # Failing to find any of the required keys
+            with self.assertRaises(KeyError):
+                tester({})
+        tester = MatchedRefCoaddMetric(**kwargs)
         with self.assertRaises(ValueError):
-            tester({})
-        tester = MatchedRefCoaddMetric(name_prefix="")
-        with self.assertRaises(ValueError):
-            tester({})
-        tester = MatchedRefCoaddMetric(unit="", name_prefix="")
+            tester.validate()
+
         tester.finalize()
-        self.assertGreater(len(list(tester.getInputSchema())), 0)
+        # This works, although it leaves none keys in the schema
+        # Should maybe be caught in populatePrepFromProcess
```

Comment:
No change needed, but just a comment that, from talking to Nate, I got the impression that it's inevitable that some things might not get set until `finalize`.

Reply:
The diffMatched tools aren't using finalize yet, but they will be once DM-36446 is merged and I'll review the tests on that ticket (amongst many other things). In this case I made this comment because it's actually subclasses of this tool that set the relevant configs so it's somewhat inevitable that the base class won't quite work correctly on its own.